main.ts

```diff
@@ -0,0 +1,321 @@
+    mainWindow.hide();
+  });
+
+  mainWindow.on("close", (event: Event) => {
+    event.preventDefault();
+    mainWindow.hide();
+    return false;
+  });
+}
+
+// This method will be called when Electron has finished
+// initialization and is ready to create browser windows.
+// Some APIs can only be used after this event occurs.
+app.on("ready", createWindow);
+
+// Quit when all windows are closed.
+app.on("window-all-closed", () => {
+  // On OS X it is common for applications and their menu bar
+  // to stay active until the user quits explicitly with Cmd + Q
+  if (process.platform !== "darwin") {
+    app.quit();
+  }
+});
+
+app.on("activate", () => {
+  // On OS X it"s common to re-create a window in the app when the
+  // dock icon is clicked and there are no other windows open.
+  if (mainWindow === null) {
+    createWindow();
+  }
+});
+
+/**
+ * appends a string to the app"s title
+ *
+ * @author Flo Dörr <flo@dörr.site>
+ */
+ipcMain.on("appendTitle", (event: Event, arg: string) => {
+  mainWindow.setTitle(`${APP_NAME}\t${arg}`);
+});
+
+/**
+ * sets an image as tray icon
+ *
+ * @author Flo Dörr <flo@dörr.site>
+ */
+
+const options: any = {
+  encoding: null,
+  method: "get",
+  url,
+};
+
+ipcMain.on("setTrayImage", (event: Event, arg: any) => {
+  options.url = arg;
+  request(options, (err: Error, resp: any, body: Buffer) => {
+    if (!err) {
+      tray.setImage(nativeImage.createFromBuffer(body));
+    } else {
+      tray.setImage(imageLocation);
+    }
+  });
+});
+
+/**
+ * is executed when electron is done loading
+ * the following code is going to register the media keys
+ *
+ * @author Flo Dörr <flo@dörr.site>
+ */
+app.on("ready", () => {
+  if (globalShortcut.register("mediaplaypause", playAndPause)) {
+    // tslint:disable-next-line:no-console
+    console.log("mediaplaypause registered!");
+  } else {
+    // tslint:disable-next-line:no-console
+    console.log("mediaplaypause NOT registered!");
+  }
+
+  if (globalShortcut.register("medianexttrack", nextTrack)) {
+    // tslint:disable-next-line:no-console
+    console.log("medianexttrack registered!");
+  } else {
+    // tslint:disable-next-line:no-console
+    console.log("medianexttrack NOT registered!");
+  }
+
+  if (globalShortcut.register("mediaprevioustrack", previousTrack)) {
+    // tslint:disable-next-line:no-console
+    console.log("mediaprevioustrack registered!");
+  } else {
+    // tslint:disable-next-line:no-console
+    console.log("mediaprevioustrack NOT registered!");
+  }
+
+  if (globalShortcut.register("mediastop", playAndPause)) {
+    // tslint:disable-next-line:no-console
+    console.log("mediastop registered!");
+  } else {
+    // tslint:disable-next-line:no-console
+    console.log("mediastop NOT registered!");
+  }
+});
+
+/**
+ * handels the "play and pause" key
+ *
+ * @author Flo Dörr <flo@dörr.site>
+ */
+function playAndPause() {
+  mainWindow.webContents.send("playAndPause");
+}
+
+/**
+ * handels "the next track" key
```

Comment:
The evil `handels` again 😄

Reply:
copy and paste is the evel :tongue: --> 432bb85